acestep/llm_inference.py

```
@@ -720,19 +741,23 @@ def initialize(
                         if hasattr(torch.cuda, "mem_get_info"):
-                            free_bytes, _ = torch.cuda.mem_get_info()
+                            free_bytes, _ = torch.cuda.mem_get_info(device_index)
                             free_gb = free_bytes / (1024**3)
                         else:
-                            total_bytes = torch.cuda.get_device_properties(0).total_memory
-                            free_gb = (total_bytes - torch.cuda.memory_reserved(0)) / (1024**3)
+                            total_bytes = torch.cuda.get_device_properties(device_index).total_memory
+                            free_gb = (
+                                total_bytes - torch.cuda.memory_reserved(device_index)
+                            ) / (1024**3)
                     except Exception:
```

Ruff (0.15.20)

[warning] Do not catch blind exception: `Exception`

(BLE001)